phpstan.neon

```diff
@@ -0,0 +1,100 @@
+includes:
+	- vendor/nunomaduro/larastan/extension.neon
+	- vendor/phpstan/phpstan-deprecation-rules/rules.neon
+	- vendor/thecodingmachine/phpstan-safe-rule/phpstan-safe-rule.neon
+	- vendor/phpstan/phpstan-strict-rules/rules.neon
+
+services:
+	-
+		class: PHPStan\Rules\StrictCalls\StrictFunctionCallsRule
+		tags:
+			- phpstan.rules.rule
+
+	-
+		class: PHPStan\Rules\SwitchConditions\MatchingTypeInSwitchCaseConditionRule
+		tags:
+			- phpstan.rules.rule
+
+parameters:
+	level: 7
+	reportMaybesInPropertyPhpDocTypes: false
+	checkMissingIterableValueType: false
+	paths:
+		- app
+	excludePaths:
+		- app/View/*
+		- app/Http/Livewire/*
+	ignoreErrors:
+		- '#Short ternary operator is not allowed#' # we like short ternary
+		- '#Construct empty\(\) is not allowed#' # we like empty()
+
+		# bunch of false positives from Eloquent
+		- '#Dynamic call to static method Illuminate\\Database\\Eloquent\\.*::insert\(\).#'
```

Comment:
I had a look into that. Unfortunately, we need these rules due to the buggy `larastan` ruleset. It is really amazing.

Without the `larastan` ruleset, the errors which suppress here do not occur, because the methods are proper dynamic methods and PhpStan finds them. But PhpStan complains about hundreds of undefined methods instead due to all these stupid facades and magic methods.

Instead of properly annotating their facades and classes with the additionally provided properties and methods which are magically available (i.e. via `__set`, `__get` and `__call`), Laravel publishes a  `larastan` ruleset for PhpStan which lies about the available methods to PhpStan. However, while doing so they forget two third of all other methods which are actually there.

Hence, with `larastan` PhpStan knows all the magic methods but cannot resolve the other methods anymore. WTF? This is the typical example of a workaround which introduces more bugs instead of fixing the original bug properly in the first place.

A friend once told me the following saying of one of his teachers (in German): "Es gibt Menschen, die probieren Ihr Leben lang nichts anderes als herauszufunden, wieviel Scheiße man auf einen Quadratmeter stapeln kann." In English: "There are people who spend their lives trying to figure out how much shit you can pile onto one square meter." This is so much true about the Laravel framework.